### PR TITLE
Add internal transaction submission file

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Just realized that previous PR was not completely full as we need to have those pure functions internally that handles the logic.  
This PR conforms to the sdk design in a way that everything in `api/` folder is a user facing interface functions and those in `internal/` folder are responsible on the actual logic.
more info https://github.com/aptos-labs/aptos-ts-sdk/pull/11/files#diff-a5a44d0986e0254ae62cd623aeaac609a288c9b2eabe5c741fd5b4a4f350a3ddR2

### Test Plan
`pnpm jest`

### CODE REVIEW

**Review by heliuchuan — COMMENTED** [review]
We should remove the dupe func docs in internal and have /api be source of truth?

File: src/transactions/transaction_builder/transaction_builder.ts [hunk 1]
```diff
@@ -180,12 +185,10 @@ export async function generateRawTransaction(args: {
  * When we call our `generateTransaction` function with the relevant type properties,
  * Typescript can infer the return type based on the appropriate function overload.
  */
-export async function generateTransaction(
-  args: GenerateSingleSignerRawTransactionArgs,
-): Promise<SingleSignerTransaction>;
-export async function generateTransaction(args: GenerateFeePayerRawTransactionArgs): Promise<FeePayerTransaction>;
-export async function generateTransaction(args: GenerateMultiAgentRawTransactionArgs): Promise<MultiAgentTransaction>;
-export async function generateTransaction(args: GenerateRawTransactionArgs): Promise<AnyRawTransaction>;
+export async function buildTransaction(args: GenerateSingleSignerRawTransactionArgs): Promise<SingleSignerTransaction>;
```
Comment (xbtmatt) [inline src/transactions/transaction_builder/transaction_builder.ts:188]: Much better name!

File: src/transactions/transaction_builder/transaction_builder.ts [hunk 2]
```diff
@@ -180,12 +185,10 @@ export async function generateRawTransaction(args: {
  * When we call our `generateTransaction` function with the relevant type properties,
```
Comment (xbtmatt) [inline src/transactions/transaction_builder/transaction_builder.ts:185]:
```suggestion
 * When we call our `buildTransaction` function with the relevant type properties,
```

**Review by xbtmatt — APPROVED** [review]
Agree with @heliuchuan on consolidating the docs to one section although we can do it later and I think you have a better idea of the intended structure here

lgtm, `buildTransaction` is definitely a much better name!

File: src/internal/transaction_submission.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ *  feePayerAddress?: AccountAddress
+ * }
+ * ```
+ */
+export async function generateTransaction(
+  args: { aptosConfig: AptosConfig } & GenerateTransactionInput,
+): Promise<AnyRawTransaction> {
+  const { aptosConfig, sender, data, options, secondarySignerAddresses, feePayerAddress } = args;
+  const payload = await generateTransactionPayload(data);
+  const rawTransaction = await buildTransaction({
```
Comment (gregnazario) [inline src/internal/transaction_submission.ts:67]: This is different than buildTransaction, or should it also be named buildTransaction?

**Review by gregnazario — COMMENTED** [review]
Can we fix the E2E transaction submission test I had to disable?

**Review by gregnazario — APPROVED** [review]

**Comment by 0xmaayan** [comment]
> Can we fix the E2E transaction submission test I had to disable?

the whole e2e transaction submission tests will get changed

**Pull request merged**